src/compiler.jl

```diff
@@ -3069,40 +3069,214 @@ function julia_error(cstr::Cstring, val::LLVM.API.LLVMValueRef, errtype::API.Err
 
-    mem = call!(B, mallocF, [mul!(B, Count, Align)])
-    LLVM.API.LLVMAddCallSiteAttribute(mem, LLVM.API.LLVMAttributeReturnIndex, EnumAttribute("noalias"; ctx))
-    LLVM.API.LLVMAddCallSiteAttribute(mem, LLVM.API.LLVMAttributeReturnIndex, EnumAttribute("nonnull"; ctx))
+    Size = nuwmul!(B, Count, AlignedSize) # should be nsw, nuw
+
+    if any_jltypes(LLVMType)
+        T_int32 = LLVM.Int32Type(ctx)
+        T_int64 = LLVM.Int64Type(ctx)
+        T_jlvalue = LLVM.StructType(LLVM.LLVMType[]; ctx)
+        T_prjlvalue = LLVM.PointerType(T_jlvalue, Tracked)
+        T_pprjlvalue = LLVM.PointerType(T_prjlvalue)
+        T_prjlvalue_UT = LLVM.PointerType(T_jlvalue)
+        T_ppjlvalue = LLVM.PointerType(LLVM.PointerType(T_jlvalue))
+
+        TT = tape_type(LLVMType)
+        if sizeof(TT) != convert(Int, AlignedSize)
+            @safe_show LLVMType
+            GPUCompiler.@safe_warn "Enzyme aligned size and Julia size disagree" AlignedSize=convert(Int, AlignedSize) sizeof(TT) fieldtypes(TT)
+        end
+        @assert sizeof(TT) == convert(Int, AlignedSize)
+        if Count isa LLVM.ConstantInt
+            N = convert(Int, Count)
+
+            ETT = EnzymeTape{N, TT}
+            @assert sizeof(ETT) <= N*convert(Int, AlignedSize)
+
+            # Obtain tag
+            tag = LLVM.ConstantInt(reinterpret(Int, Base.pointer_from_objref(ETT)); ctx)  # do we need to root ETT
+            tag = LLVM.const_inttoptr(tag, T_prjlvalue_UT)
+            tag = LLVM.const_addrspacecast(tag, T_prjlvalue)
+        else
+            # TODO: Version switch to get jl_box_int64 and not call37...
+            box_int64 = get_function!(mod, "ijl_box_int64", LLVM.FunctionType(T_prjlvalue, [T_int64]))
+            boxed_count = call!(B, box_int64, [Count])
+
+            f_apply_type = get_function!(mod, "jl_f_apply_type", LLVM.FunctionType(T_prjlvalue, [T_prjlvalue, T_pprjlvalue, T_int32]))
+
+            FT = LLVM.FunctionType(T_prjlvalue, [T_prjlvalue, T_prjlvalue, T_prjlvalue, T_prjlvalue])
+            f_apply_type = bitcast!(B, f_apply_type, LLVM.PointerType(FT))
+
+            EnzymeTapeT = LLVM.ConstantInt(reinterpret(Int, Base.pointer_from_objref(base_type(EnzymeTape))); ctx)
+            EnzymeTapeT = LLVM.const_inttoptr(EnzymeTapeT, T_prjlvalue_UT)
+            EnzymeTapeT = LLVM.const_addrspacecast(EnzymeTapeT, T_prjlvalue)
+
+            TapeT = LLVM.ConstantInt(reinterpret(Int, Base.pointer_from_objref(TT)); ctx)
+            TapeT = LLVM.const_inttoptr(TapeT, T_prjlvalue_UT)
+            TapeT = LLVM.const_addrspacecast(TapeT, T_prjlvalue)
+
+            # call cc37 nonnull {}* bitcast ({}* ({}*, {}**, i32)* @jl_f_apply_type to {}* ({}*, {}*, {}*, {}*)*)({}* null, {}* inttoptr (i64 140150176657296 to {}*), {}* %4, {}* inttoptr (i64 140149987564368 to {}*))
+            tag = call!(B, f_apply_type, [LLVM.PointerNull(T_prjlvalue), EnzymeTapeT, boxed_count, TapeT])
+            LLVM.callconv!(tag, 37)
```

Comment:
For 1.9+ this needs to be upgraded to a julia.call or julia.call2

Reply:
@vchuravy I'm fine merging without this fix, but bumping here.

Reply:
I would like to fix 1.9 support since that is the only thing that uses that branch